src/api/consumer/lib/add.js

```diff
@@ -35,13 +38,19 @@ export async function addMany(
     componentDefinition.exclude = componentDefinition.exclude
       ? componentDefinition.exclude.map(excludeFile => path.normalize(excludeFile.trim()))
       : [];
+    componentDefinition.configuredConsumer = configureConsumer;
     const addComponents = new AddComponents(consumer, componentDefinition);
     addComponentsArr.push(addComponents);
   });
   const addResults = [];
   await Promise.all(
     addComponentsArr.map(async function (addComponents) {
-      const addResultsSingle = await addComponents.add();
+      let addResultsSingle;
+      try {
+        addResultsSingle = await addComponents.add();
+      } catch (ex) {
+        logger.error(`got the following exception while adding componnet ${ex.toString()}`);
```

Comment:
I'm not sure we want to catch it.. i think it's better to do all or nothing.
You can maybe add a param which defines if you allow partial success